

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Building chimeras &mdash; ProtLego 1.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Structural analysis of chimeras" href="structural.html" />
    <link rel="prev" title="Drawing networks" href="network.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> ProtLego
          

          
            
            <img src="_static/logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Code</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="builder.html">Building Chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="builder.html#module-builder.chimera">Chimera Class</a></li>
<li class="toctree-l1"><a class="reference internal" href="database.html">Fetching hits from Fuzzle</a></li>
<li class="toctree-l1"><a class="reference internal" href="networks.html">Networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html">Clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#module-structural.hh_networks">HH-networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#module-structural.salt_bridges">Salt-bridges</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#module-structural.analysis">Further structural analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">User Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html#pip-nstallation">PIP nstallation</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html#conda-installation">CONDA installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html#setup">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="fuzzle.html">Visit Fuzzle</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction to Jupyter-notebooks and Python primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="fetching.html">Fetching from the Fuzzle database</a></li>
<li class="toctree-l1"><a class="reference internal" href="network.html">Drawing networks</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Building chimeras</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Chimera-Building-using-Protlego">Chimera-Building using Protlego</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="structural.html">Structural analysis of chimeras</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ProtLego</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Building chimeras</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/building.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Building-chimeras">
<h1>Building chimeras<a class="headerlink" href="#Building-chimeras" title="Permalink to this headline">¶</a></h1>
<p>We can build chimeras from two proteins that present a fragment in common. There are several ways to do this, but we will here build a chimera by aligning the domains (in the fragment they have in common) and finding a crossover point in this region that divides the two parents in two parts. The parts are then shuffled and recombined.</p>
<p>First import the protlego module.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">protlego.all</span> <span class="kn">import</span> <span class="o">*</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
/agh/projects/noelia/github/protlego/TMalign
</pre></div></div>
</div>
<p>Fetch for example by the HitID from the Fuzzle database, as explained more in detail in the “Fetching from the Fuzzle database” section. We are fetching a hit between the P-loop and the Rossmann folds in this case.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">myhit</span><span class="o">=</span> <span class="n">fetch_id</span><span class="p">(</span><span class="s1">&#39;4413706&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="Chimera-Building-using-Protlego">
<h2>Chimera-Building using Protlego<a class="headerlink" href="#Chimera-Building-using-Protlego" title="Permalink to this headline">¶</a></h2>
<p>First create an instance of the Builder class.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">a</span><span class="o">=</span><span class="n">Builder</span><span class="p">(</span><span class="n">myhit</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-16 12:17:21,548 - protlego - INFO - Attempting to download domain d2dfda1 from the SCOP server
2020-10-16 12:17:22,721 - protlego - INFO - File downloaded as /tmp/tmp723odzdf.pdb
2020-10-16 12:17:22,721 - protlego - INFO - Attempting to download domain d1wa5a_ from the SCOP server
2020-10-16 12:17:23,881 - protlego - INFO - File downloaded as /tmp/tmpz50rsvv3.pdb
2020-10-16 12:17:23,883 - protlego - INFO - Loading /tmp/tmp723odzdf.pdb as a chimera object
2020-10-16 12:17:23,960 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=&#34;all&#34; option when reading the file.
2020-10-16 12:17:23,964 - protlego - INFO - Loading /tmp/tmpz50rsvv3.pdb as a chimera object
</pre></div></div>
</div>
<p>You can see the query and the subject of the hit stored in the variables qPDB and sPDB as chimera objects</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">a</span><span class="o">.</span><span class="n">qPDB</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>&lt;protlego.builder.chimera.Chimera object at 0x7f2f9ff12b70&gt;
Protein with:
Number of residues: 145
Number of atoms: 1054
</pre></div>
</div>
</div>
<p>As described in the previous tutorial, myhit includes information about the two proteins that share a fragment, like the scop IDs, length, and TM-align scores, among others.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="nb">print</span><span class="p">(</span><span class="n">myhit</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">q_scop_id</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">s_scop_id</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">cols</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">score_tm</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Hit between d2dfda1 and d1wa5a_ with probability 81.7 %

c.2.1.5
c.37.1.8
101
0.54938
</pre></div></div>
</div>
<p>Align the sequences of query and subject with HHPred using get_alignment function.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">aln</span><span class="o">=</span><span class="n">a</span><span class="o">.</span><span class="n">get_alignment</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">query</span><span class="p">,</span><span class="n">myhit</span><span class="o">.</span><span class="n">no</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-16 12:17:24,046 - protlego - INFO - Attempting to download hhs file for d2dfda1 from the FUZZLE server
2020-10-16 12:17:24,184 - protlego - INFO - File downloaded as /tmp/tmpg1t9myzv.hhs
</pre></div></div>
</div>
<p>Now we can have a look at the sequence alignment:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="nb">print</span><span class="p">(</span><span class="n">aln</span><span class="o">.</span><span class="n">query</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">aln</span><span class="o">.</span><span class="n">subject</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
KVAVLGASGGIGQ-PLSLLLKNS------------PLVSRLTLYDIAHTPGVAADLSHIETKAAVKGYLGPEQLPDCLKGCDVVVIPAGVPRKPGMTRDDLFNTNATIVATLTAACAQHCPEAMICVIANPVN
KLVLVG-DGGTGKTTFVKRHLTGEFEKKYVATLGVEVHPLVFHTNRGPIKFNVWDTAGQEKFGGLRDGY--------YIQAQCAIIMFDVTSRV----------TYKNVPNWHRDLVRVCENIPIVLCGNKVD
</pre></div></div>
</div>
<p>The sequence alignment corresponds to the exact fragment that the two proteins have in common, so it does not include the full protein sequences. Now, we need to map these alignments to the protein structures, because ultimately we need to align the two protein structures in these fragment region.</p>
<p>We can thus superimpose the structures with the superimpose_structures() function, and we choose to align the structures using partial alignment. This means that instead of only one, global alignment being produced, rather, several alignments comprising smaller regions of the fragment (10-20 residues) are undertaken in parallel. We do this to provide multiple regions where the two proteins are closer than 1 armstrong in distance, and thus obtain more fusion points for the chimeras.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">qPDB</span><span class="p">,</span> <span class="n">sPDB</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">superimpose_structures</span><span class="p">(</span><span class="n">aln</span><span class="p">,</span><span class="n">partial_alignment</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-16 12:17:25,126 - protlego - INFO - Performing alignment 1 with TMalign
/agh/projects/noelia/miniconda3/envs/release/lib/python3.6/site-packages/moleculekit/align.py:16: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, A), array(float32, 2d, A))
  covariance = np.dot(P.T, Q)
/agh/projects/noelia/miniconda3/envs/release/lib/python3.6/site-packages/moleculekit/align.py:54: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, A), array(float32, 2d, A))
  all1 = np.dot(all1, rot.T)
2020-10-16 12:17:28,332 - protlego - INFO - The RMSD between the fragments is 0.6881077885627747 over 6 alpha carbons
2020-10-16 12:17:28,340 - protlego - INFO - Performing alignment 2 with TMalign
2020-10-16 12:17:28,420 - protlego - INFO - The RMSD between the fragments is 2.323467254638672 over 6 alpha carbons
2020-10-16 12:17:28,430 - protlego - INFO - Performing alignment 3 with TMalign
2020-10-16 12:17:28,498 - protlego - INFO - The RMSD between the fragments is 1.180463433265686 over 9 alpha carbons
2020-10-16 12:17:28,506 - protlego - INFO - Performing alignment 4 with TMalign
2020-10-16 12:17:28,606 - protlego - INFO - The RMSD between the fragments is 10.348337173461914 over 34 alpha carbons
2020-10-16 12:17:28,616 - protlego - INFO - Performing alignment 5 with TMalign
2020-10-16 12:17:28,708 - protlego - INFO - The RMSD between the fragments is 4.646137714385986 over 17 alpha carbons
2020-10-16 12:17:28,717 - protlego - INFO - Performing alignment 6 with TMalign
2020-10-16 12:17:28,797 - protlego - INFO - The RMSD between the fragments is 3.5291929244995117 over 29 alpha carbons
</pre></div></div>
</div>
<p>The two variables produced are two dictionaries, which contain several molecules, each pair in a geometry that optimizes each of the alignments. We can visualize these structures easily, by simply calling the view() method. A new VMD window will open showing the structure. For example, less visualize the first alignment by slicing the first key in the dictionary:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">qPDB</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<p>The superimpose_structures() command also produces two lists for query and subject. One of them is global_qpairs, which simply contains the indices of the protein that correspond to the srquence aligment. This list is used when we superimpose the structures using a global alignment (all atoms taken into account at once). The other is qpairs, which correspond to the same residues, but they are clustered in smaller regions that will be aligned iteratively if we choose to perform a partial
alignment.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="nb">print</span><span class="p">(</span><span class="n">a</span><span class="o">.</span><span class="n">qpairs</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">a</span><span class="o">.</span><span class="n">global_qpairs</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
[[14, 23, 30, 35, 42, 50], [59, 65, 69, 73, 81, 85], [94, 101, 109, 115, 123, 131, 139, 148, 156], [162, 169, 177, 184, 190, 201, 209, 216, 224, 236, 244, 252, 257, 267, 274, 281, 285, 292, 297, 302, 310, 318, 324, 334, 342, 351, 358, 364, 369, 374, 381, 390, 394, 406], [472, 480, 489, 493, 499, 507, 514, 521, 528, 536, 543, 548, 552, 559, 566, 577, 586], [673, 681, 686, 693, 701, 708, 713, 720, 728, 735, 740, 745, 751, 756, 765, 775, 781, 788, 797, 802, 810, 818, 824, 831, 839, 844, 852, 859, 866]]
[[14, 23, 30, 35, 42, 50, 59, 65, 69, 73, 81, 85, 94, 101, 109, 115, 123, 131, 139, 148, 156, 162, 169, 177, 184, 190, 201, 209, 216, 224, 236, 244, 252, 257, 267, 274, 281, 285, 292, 297, 302, 310, 318, 324, 334, 342, 351, 358, 364, 369, 374, 381, 390, 394, 406, 472, 480, 489, 493, 499, 507, 514, 521, 528, 536, 543, 548, 552, 559, 566, 577, 586, 673, 681, 686, 693, 701, 708, 713, 720, 728, 735, 740, 745, 751, 756, 765, 775, 781, 788, 797, 802, 810, 818, 824, 831, 839, 844, 852, 859, 866]]
</pre></div></div>
</div>
<p>Now we can build all the possible chimeras and store them in a dictionary.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">chimeras</span><span class="o">=</span><span class="n">a</span><span class="o">.</span><span class="n">build_chimeras</span><span class="p">(</span><span class="n">partial_alignment</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">chimeras</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>{&#39;comb1_72&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99764f28&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1278,
 &#39;comb1_73&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f997614e0&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1279,
 &#39;comb1_74&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99769c18&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1274,
 &#39;comb2_75&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f998b8fd0&gt;
 Protein with:
 Number of residues: 152
 Number of atoms: 1166,
 &#39;comb1_76&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f9973ffd0&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1274,
 &#39;comb2_76&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f998c7fd0&gt;
 Protein with:
 Number of residues: 152
 Number of atoms: 1167,
 &#39;comb1_77&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99812fd0&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1275,
 &#39;comb1_78&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99759fd0&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1277,
 &#39;comb1_79&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99827b00&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1276,
 &#39;comb2_80&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f9975a048&gt;
 Protein with:
 Number of residues: 152
 Number of atoms: 1165,
 &#39;comb1_81&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99735fd0&gt;
 Protein with:
 Number of residues: 163
 Number of atoms: 1275,
 &#39;comb2_81&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99757080&gt;
 Protein with:
 Number of residues: 152
 Number of atoms: 1166,
 &#39;comb2_103&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f9981d048&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1101,
 &#39;comb2_104&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99823208&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1103,
 &#39;comb2_105&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f997f8048&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1104,
 &#39;comb2_106&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f997fafd0&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1110,
 &#39;comb2_107&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99ba1940&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1113,
 &#39;comb2_109&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99b77fd0&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1122,
 &#39;comb2_112&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99b85ba8&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1128,
 &#39;comb2_113&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99b9e7f0&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1125,
 &#39;comb2_118&#39;: &lt;protlego.builder.chimera.Chimera object at 0x7f2f99b8efd0&gt;
 Protein with:
 Number of residues: 142
 Number of atoms: 1128}
</pre></div>
</div>
</div>
<p>From the Builder class you can plot all the possible chimeras, showing the distance on the y-axis and the fragment of the query relative to the domain in the subject. There are five possible scenarios. For every two fussion point that is under the cutoff distanc (1 Armstrong in this case) we can build chimeras starting from the N-terminus and C-terminus. When the two resulting chimeras at this point present backbone clashes, the plot shows a black dot at this position. If the N-terminus or
C-terminus is chimera is buildable, we show that position with a cross in yellow, and red, respectively. In some fusion points, like position 145 in this example, it is possible to build both N-terminus and C-terminus chimeras.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">a</span><span class="o">.</span><span class="n">plot_curves</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">query</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/building_28_0.png" src="_images/building_28_0.png" />
</div>
</div>
<p>To access the chimeras we can retrieve them from the key value in the dictionary. Combinations starting with 1 come from the query, and combinations starting with 2 come from the subject.</p>
<p>To view or write the structures out you can store the chimera from the chimeras dictionary in a variable and use the view and the write function.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">mychimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">mychimera</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<p>When we represent the structures we can see the part that comes from each parent is shown in different colors. We can also save the structure as a pdb anytime:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">mychimera</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;./mychimera.pdb&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>We can, in fact, visualize all of them at once by looping over the structures:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">chimera</span> <span class="ow">in</span> <span class="n">chimeras</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">chimera</span><span class="o">.</span><span class="n">view</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="n">name</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Note that VMD guesses the bonds between atoms by computing distances. As these structures have not been minimized they do not have idealized bond distances and sometimes VMD cannot guess properly the NewCartoon representation. Choose Licorice Method to represent those cases.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="structural.html" class="btn btn-neutral float-right" title="Structural analysis of chimeras" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="network.html" class="btn btn-neutral float-left" title="Drawing networks" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Noelia Ferruz

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>