

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Clusters &mdash; ProtLego 1.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installation" href="installation.html" />
    <link rel="prev" title="Energy scoring" href="energy.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> ProtLego
          

          
            
            <img src="_static/logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Code</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="builder.html">Building Chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="builder.html#module-builder.chimera">Chimera Class</a></li>
<li class="toctree-l1"><a class="reference internal" href="database.html">Fetching hits from Fuzzle</a></li>
<li class="toctree-l1"><a class="reference internal" href="networks.html">Networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="#module-structural.hh_networks">HH-networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="#module-structural.salt_bridges">Salt-bridges</a></li>
<li class="toctree-l1"><a class="reference internal" href="#module-structural.analysestrc">Further structural analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">User Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fuzzle.html">Visit Fuzzle</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction to Jupyter-notebooks and Python primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="fetching.html">Fetching from the Fuzzle database</a></li>
<li class="toctree-l1"><a class="reference internal" href="network.html">Drawing networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="building.html">Building chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="structural.html">Structural analysis of chimeras</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ProtLego</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Clusters</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/analyse.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="module-structural.clusters">
<span id="clusters"></span><h1>Clusters<a class="headerlink" href="#module-structural.clusters" title="Permalink to this headline">¶</a></h1>
<dl class="class">
<dt id="structural.clusters.Atom">
<em class="property">class </em><code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">Atom</code><span class="sig-paren">(</span><em class="sig-param">index: int</em>, <em class="sig-param">mol: moleculekit.molecule.Molecule</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/clusters.html#Atom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.Atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Class to handle ILV atoms
index: the vmd atom index
mol: an htmd.htmd.molecule object</p>
<dl class="method">
<dt id="structural.clusters.Atom.get_neighbors">
<code class="sig-name descname">get_neighbors</code><span class="sig-paren">(</span><em class="sig-param">mol: moleculekit.molecule.Molecule</em><span class="sig-paren">)</span> &#x2192; numpy.array<a class="reference internal" href="_modules/structural/clusters.html#Atom.get_neighbors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.Atom.get_neighbors" title="Permalink to this definition">¶</a></dt>
<dd><p>Provides all indices of atoms within 6.56 A of this atom.
6.56 is the upper bound of a possible neighbor 1.88 (C) + 1.4 + 1.4 + 1.88 (C).</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="structural.clusters.Cluster">
<em class="property">class </em><code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">Cluster</code><span class="sig-paren">(</span><em class="sig-param">area</em>, <em class="sig-param">residues</em>, <em class="sig-param">contacts</em>, <em class="sig-param">ratio_contacts_residue</em>, <em class="sig-param">ratio_area_residue</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/clusters.html#Cluster"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.Cluster" title="Permalink to this definition">¶</a></dt>
<dd><dl class="method">
<dt id="structural.clusters.Cluster.area">
<em class="property">property </em><code class="sig-name descname">area</code><a class="headerlink" href="#structural.clusters.Cluster.area" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 0</p>
</dd></dl>

<dl class="method">
<dt id="structural.clusters.Cluster.contacts">
<em class="property">property </em><code class="sig-name descname">contacts</code><a class="headerlink" href="#structural.clusters.Cluster.contacts" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 2</p>
</dd></dl>

<dl class="method">
<dt id="structural.clusters.Cluster.ratio_area_residue">
<em class="property">property </em><code class="sig-name descname">ratio_area_residue</code><a class="headerlink" href="#structural.clusters.Cluster.ratio_area_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 4</p>
</dd></dl>

<dl class="method">
<dt id="structural.clusters.Cluster.ratio_contacts_residue">
<em class="property">property </em><code class="sig-name descname">ratio_contacts_residue</code><a class="headerlink" href="#structural.clusters.Cluster.ratio_contacts_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 3</p>
</dd></dl>

<dl class="method">
<dt id="structural.clusters.Cluster.residues">
<em class="property">property </em><code class="sig-name descname">residues</code><a class="headerlink" href="#structural.clusters.Cluster.residues" title="Permalink to this definition">¶</a></dt>
<dd><p>Alias for field number 1</p>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="structural.clusters.add_clusters">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">add_clusters</code><span class="sig-paren">(</span><em class="sig-param">mol</em>, <em class="sig-param">g: graph_tool.Graph</em>, <em class="sig-param">components: graph_tool.PropertyArray</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/clusters.html#add_clusters"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.add_clusters" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mol</strong> – </p></li>
<li><p><strong>g</strong> – </p></li>
<li><p><strong>components</strong> – </p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Molecule representations
A list with Cluster objects</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.create_graph">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">create_graph</code><span class="sig-paren">(</span><em class="sig-param">resid_matrix: numpy.array</em>, <em class="sig-param">resid_list: numpy.array</em>, <em class="sig-param">cutoff_area: float = 10.0</em><span class="sig-paren">)</span> &#x2192; graph_tool.Graph<a class="reference internal" href="_modules/structural/clusters.html#create_graph"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.create_graph" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>resid_matrix</strong> – the ILVresid x ILVresid area matrix</p></li>
<li><p><strong>resid_list</strong> – the index x index area matrix</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A Graph object where each component is a ILV cluster</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.fill_matrices">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">fill_matrices</code><span class="sig-paren">(</span><em class="sig-param">atom: structural.clusters.Atom</em>, <em class="sig-param">mol: moleculekit.molecule.Molecule</em>, <em class="sig-param">atom_matrix: numpy.array</em>, <em class="sig-param">resid_matrix: numpy.array</em>, <em class="sig-param">indices: numpy.array</em>, <em class="sig-param">resids: numpy.array</em><span class="sig-paren">)</span> &#x2192; Tuple[numpy.array, numpy.array]<a class="reference internal" href="_modules/structural/clusters.html#fill_matrices"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.fill_matrices" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atom</strong> – An Atom class</p></li>
<li><p><strong>mol</strong> – an htmd.htmd.molecule object</p></li>
<li><p><strong>atom_matrix</strong> – the index x index area matrix</p></li>
<li><p><strong>resid_matrix</strong> – the ILVresid x ILVresid area matrix</p></li>
<li><p><strong>indices</strong> – the indices that belong to ILV sidechain heavy atoms</p></li>
<li><p><strong>resids</strong> – the resids that belong to ILV sidechain heavy atoms</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Updated atom_matrix and resid_matrix</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.filter_mol">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">filter_mol</code><span class="sig-paren">(</span><em class="sig-param">inputmolfile: str</em><span class="sig-paren">)</span> &#x2192; moleculekit.molecule.Molecule<a class="reference internal" href="_modules/structural/clusters.html#filter_mol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.filter_mol" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads, filters the object to chain A and writes filtered pdb out.
:param inputmolfile: path to the pdb file
:return: the Molecule object</p>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.generate_sphere_points">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">generate_sphere_points</code><span class="sig-paren">(</span><em class="sig-param">coords: numpy.array</em>, <em class="sig-param">n: int = 610</em>, <em class="sig-param">radius: float = 1.88</em><span class="sig-paren">)</span> &#x2192; numpy.array<a class="reference internal" href="_modules/structural/clusters.html#generate_sphere_points"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.generate_sphere_points" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>coords</strong> – The coordinates of the center of the atom</p></li>
<li><p><strong>n</strong> – number of points to sample</p></li>
<li><p><strong>radius</strong> – the radius of the atom</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>a nx3 vector with the point coordinates</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.postprocess_session">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">postprocess_session</code><span class="sig-paren">(</span><em class="sig-param">inputmolfile: str</em>, <em class="sig-param">outputname: str</em><span class="sig-paren">)</span> &#x2192; None<a class="reference internal" href="_modules/structural/clusters.html#postprocess_session"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.postprocess_session" title="Permalink to this definition">¶</a></dt>
<dd><p>Modifies the VMD session to not include tmp files
:param outputname: The vmd session (output file)
:param inputmolfile: Path to the pdb file already processed (filtered and or protonated)
:return: None. Modifies the file inline</p>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.retrieve_indices">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">retrieve_indices</code><span class="sig-paren">(</span><em class="sig-param">matrix: numpy.array</em>, <em class="sig-param">coords: numpy.array</em>, <em class="sig-param">neighborpositions: numpy.array</em>, <em class="sig-param">radius: float = 1.88</em><span class="sig-paren">)</span> &#x2192; numpy.array<a class="reference internal" href="_modules/structural/clusters.html#retrieve_indices"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.retrieve_indices" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes if each of the n sphere points are penetrating neighboring spheres
:param matrix: n x m Distance matrix where n is the number of sphere points and m the number of neighbors
:param coords: the coordinates of the atom
:param neighborpositions: Coordinates of the neighbors
:param radius: radius of the atom
:return: The atoms that are in closest with each n sphere points.</p>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.retrieve_neighbor_positions">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">retrieve_neighbor_positions</code><span class="sig-paren">(</span><em class="sig-param">atom: structural.clusters.Atom</em>, <em class="sig-param">mol: moleculekit.molecule.Molecule</em><span class="sig-paren">)</span> &#x2192; Tuple[numpy.array, Dict[int, int]]<a class="reference internal" href="_modules/structural/clusters.html#retrieve_neighbor_positions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.retrieve_neighbor_positions" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atom</strong> – an Atom object</p></li>
<li><p><strong>mol</strong> – a htmd.htmd.molecule object</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A tuple object with the positions of the neighboring atoms. A dictionary indexing column positions to resid positions</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.write_clusters">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">write_clusters</code><span class="sig-paren">(</span><em class="sig-param">g: graph_tool.Graph</em>, <em class="sig-param">components: graph_tool.PropertyArray</em>, <em class="sig-param">inputmolfile: str</em>, <em class="sig-param">outputname: str</em><span class="sig-paren">)</span> &#x2192; None<a class="reference internal" href="_modules/structural/clusters.html#write_clusters"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.write_clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>This function prints the clusters to the terminal and outputs them
into a VMD session
:param g: A Graph object
:param outputname: The pdb filename.
:param outputname: The file name to output the VMD session to.
:return:</p>
</dd></dl>

<dl class="function">
<dt id="structural.clusters.write_largest_cluster">
<code class="sig-prename descclassname">structural.clusters.</code><code class="sig-name descname">write_largest_cluster</code><span class="sig-paren">(</span><em class="sig-param">g: graph_tool.Graph</em>, <em class="sig-param">inputmolfile: str</em>, <em class="sig-param">outputname: str</em><span class="sig-paren">)</span> &#x2192; None<a class="reference internal" href="_modules/structural/clusters.html#write_largest_cluster"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.clusters.write_largest_cluster" title="Permalink to this definition">¶</a></dt>
<dd><p>This function prints the clusters to the terminal and outputs them
into a VMD session
:param g: A Graph object
:param outputname: The pdb filename.
:param outputname: The file name to output the VMD session to.
:return:</p>
</dd></dl>

</div>
<div class="section" id="module-structural.hh_networks">
<span id="hh-networks"></span><h1>HH-networks<a class="headerlink" href="#module-structural.hh_networks" title="Permalink to this headline">¶</a></h1>
<dl class="function">
<dt id="structural.hh_networks.add_networks">
<code class="sig-prename descclassname">structural.hh_networks.</code><code class="sig-name descname">add_networks</code><span class="sig-paren">(</span><em class="sig-param">mol</em>, <em class="sig-param">g: graph_tool.Graph</em>, <em class="sig-param">components: graph_tool.PropertyArray</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/hh_networks.html#add_networks"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.hh_networks.add_networks" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mol</strong> – the Chimera where to add the representations to.</p></li>
<li><p><strong>g</strong> – a graph-tool graph object with the networks property</p></li>
<li><p><strong>components</strong> – the components from the graph</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>INcludes the representations in the Chimera object.</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.hh_networks.make_graph_hh">
<code class="sig-prename descclassname">structural.hh_networks.</code><code class="sig-name descname">make_graph_hh</code><span class="sig-paren">(</span><em class="sig-param">hbonds: numpy.array</em>, <em class="sig-param">trajectory: mdtraj.core.trajectory.Trajectory</em><span class="sig-paren">)</span> &#x2192; graph_tool.Graph<a class="reference internal" href="_modules/structural/hh_networks.html#make_graph_hh"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.hh_networks.make_graph_hh" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>hbonds</strong> – a np.array indicating indexes of atoms that have an h-bond</p></li>
<li><p><strong>trajectory</strong> – an mdtraj trajectory. It can also be a pdb file processed with mdtraj as trajectory</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A graph-tool graph object</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="structural.hh_networks.protonate_mol">
<code class="sig-prename descclassname">structural.hh_networks.</code><code class="sig-name descname">protonate_mol</code><span class="sig-paren">(</span><em class="sig-param">inputmolfile: str</em><span class="sig-paren">)</span> &#x2192; moleculekit.molecule.Molecule<a class="reference internal" href="_modules/structural/hh_networks.html#protonate_mol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.hh_networks.protonate_mol" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads, filters the object to chain A and protonates the selection.
It then writes the filtered protonated pdb out.
:param inputmolfile: path to the pdb file
:return: a Molecule object with the correct protonation state</p>
</dd></dl>

<dl class="function">
<dt id="structural.hh_networks.write_networks">
<code class="sig-prename descclassname">structural.hh_networks.</code><code class="sig-name descname">write_networks</code><span class="sig-paren">(</span><em class="sig-param">g: graph_tool.Graph</em>, <em class="sig-param">components: graph_tool.PropertyArray</em>, <em class="sig-param">inputmolfile: str</em>, <em class="sig-param">outputname: str</em><span class="sig-paren">)</span> &#x2192; None<a class="reference internal" href="_modules/structural/hh_networks.html#write_networks"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.hh_networks.write_networks" title="Permalink to this definition">¶</a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> – A graph-tool graph object</p></li>
<li><p><strong>components</strong> – the components of the graph</p></li>
<li><p><strong>inputmolfile</strong> – the pdb of the input protein</p></li>
<li><p><strong>outputname</strong> – a path wehre to write the output</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Writes a file named outputname + “hh-networks.txt”</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-structural.salt_bridges">
<span id="salt-bridges"></span><h1>Salt-bridges<a class="headerlink" href="#module-structural.salt_bridges" title="Permalink to this headline">¶</a></h1>
<dl class="function">
<dt id="structural.salt_bridges.write_salt_bridges">
<code class="sig-prename descclassname">structural.salt_bridges.</code><code class="sig-name descname">write_salt_bridges</code><span class="sig-paren">(</span><em class="sig-param">data: numpy.ndarray</em>, <em class="sig-param">mapping: pandas.core.frame.DataFrame</em>, <em class="sig-param">mol: moleculekit.molecule.Molecule</em>, <em class="sig-param">outputname: str</em><span class="sig-paren">)</span> &#x2192; None<a class="reference internal" href="_modules/structural/salt_bridges.html#write_salt_bridges"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.salt_bridges.write_salt_bridges" title="Permalink to this definition">¶</a></dt>
<dd><p>This function outputs the salt bridges into a VMD session
:param data: A MetricDistance object
:param mapping: A DataFrame object including the index - residue mapping
:param mol: The pdb filename.
:param outputname: The file prefix name to output the VMD session to. Example: “protein2”
:return: A file with the VMD session named outputname+”-salt-bridges.txt”</p>
</dd></dl>

</div>
<div class="section" id="module-structural.analysestrc">
<span id="further-structural-analysis"></span><h1>Further structural analysis<a class="headerlink" href="#module-structural.analysestrc" title="Permalink to this headline">¶</a></h1>
<dl class="function">
<dt id="structural.analysestrc.calc_contact_order">
<code class="sig-prename descclassname">structural.analysestrc.</code><code class="sig-name descname">calc_contact_order</code><span class="sig-paren">(</span><em class="sig-param">chimera: protlego.builder.chimera.Chimera = None</em>, <em class="sig-param">filename: str = None</em>, <em class="sig-param">diss_cutoff: int = 8</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/analysestrc.html#calc_contact_order"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.analysestrc.calc_contact_order" title="Permalink to this definition">¶</a></dt>
<dd><p>The contact order of a protein is a measure of the locality of the inter-amino acid contacts in the
native folded state. It is computed as the average seqeuence distance between residues that form contacts
below a threshold in the folded protein divided by the total length of the protein”
:param chimera: A Chimera object with n residues.
:param filename: path to a pdb file
:param diss_cutoff: The maximum distance in Armstrong between two residues to be in contact, default 8 Angstroms
:return: the contact order (%)</p>
</dd></dl>

<dl class="function">
<dt id="structural.analysestrc.calc_dist_matrix">
<code class="sig-prename descclassname">structural.analysestrc.</code><code class="sig-name descname">calc_dist_matrix</code><span class="sig-paren">(</span><em class="sig-param">chimera: protlego.builder.chimera.Chimera = None</em>, <em class="sig-param">filename: str = None</em>, <em class="sig-param">selection: str = 'residue'</em>, <em class="sig-param">type='contacts'</em>, <em class="sig-param">plot=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/analysestrc.html#calc_dist_matrix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.analysestrc.calc_dist_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a matrix of C-alpha distances for a given pdb
:param chimera: A Chimera object with n residues.
:param filename: path to a pdb file
:param selection: How to compute the distance. ‘residue’ (the closest two
:param type: between contacts (contact map when distances are below 8 armstrongs) or distances atoms between two residues) or ‘alpha’ distance of the alpha carbons.
:param plot: whether to plot the distance matrix. Default is False
:return: matrix. np.array. An n by n distance matrix.</p>
</dd></dl>

<dl class="function">
<dt id="structural.analysestrc.calc_dssp">
<code class="sig-prename descclassname">structural.analysestrc.</code><code class="sig-name descname">calc_dssp</code><span class="sig-paren">(</span><em class="sig-param">chimera: protlego.builder.chimera.Chimera = None</em>, <em class="sig-param">filename: str = None</em>, <em class="sig-param">simplified: bool = True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/analysestrc.html#calc_dssp"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.analysestrc.calc_dssp" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute Dictionary of protein secondary structure (DSSP) secondary structure assignments.
This funcion uses the MDtraj compute_dssp implementation as a basis.
:param chimera: A Chimera object.
:param filename: path to a pdb file
:param simplified: Use the simplified 3-category assignment scheme. Otherwise the original 8-category scheme is used.
:return: assignments np.ndarray. The secondary structure assignment for each residue</p>
</dd></dl>

<dl class="function">
<dt id="structural.analysestrc.calc_sasa">
<code class="sig-prename descclassname">structural.analysestrc.</code><code class="sig-name descname">calc_sasa</code><span class="sig-paren">(</span><em class="sig-param">chimera: protlego.builder.chimera.Chimera = None</em>, <em class="sig-param">filename: str = None</em>, <em class="sig-param">probe_radius: float = 0.14</em>, <em class="sig-param">n_sphere_points: int = 960</em>, <em class="sig-param">sasa_type='total'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/analysestrc.html#calc_sasa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.analysestrc.calc_sasa" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the Solvent Accessible Surface Area of the protein.
This funcion uses the MDtraj shrake_rupley implementation as a basis.
:param chimera: A Chimera object.
:param filename: Path to a pdb file
:param probe_radius: The radius of the probe, in nm.
:param n_sphere_points: the number of points representing the sufrace of each atom. Higher values lead to more accuracy.
:param sasa_type: Type of calculation to perform. To select from polar, apolar, or total.
:return: areas: np.array containing the area of the chimera in Angstrom^2</p>
</dd></dl>

<dl class="function">
<dt id="structural.analysestrc.hhbond_plot">
<code class="sig-prename descclassname">structural.analysestrc.</code><code class="sig-name descname">hhbond_plot</code><span class="sig-paren">(</span><em class="sig-param">chimera: protlego.builder.chimera.Chimera = None</em>, <em class="sig-param">filename: str = None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/structural/analysestrc.html#hhbond_plot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#structural.analysestrc.hhbond_plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes a hhbond plot of a chimera object or a file. One of the two inputs
must be provided.
:param chimera: the chimera from where to compute the hydrogen bond plot
:param filename: a path where to find the pdb file.
:return: A contact map with hydrogen bond as metric</p>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="installation.html" class="btn btn-neutral float-right" title="Installation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="energy.html" class="btn btn-neutral float-left" title="Energy scoring" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Noelia Ferruz

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>