

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Structural analysis of chimeras &mdash; ProtLego 1.0.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Building chimeras" href="building.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> ProtLego
          

          
            
            <img src="_static/logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Code</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="builder.html">Building Chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="builder.html#module-builder.chimera">Chimera Class</a></li>
<li class="toctree-l1"><a class="reference internal" href="database.html">Fetching hits from Fuzzle</a></li>
<li class="toctree-l1"><a class="reference internal" href="networks.html">Networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html">Clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#module-structural.hh_networks">HH-networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#module-structural.salt_bridges">Salt-bridges</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#further-structural-analysis">Further structural analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">User Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="fuzzle.html">Visit Fuzzle</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction to Jupyter-notebooks and Python primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="fetching.html">Fetching from the Fuzzle database</a></li>
<li class="toctree-l1"><a class="reference internal" href="network.html">1. Drawing relationships between two superfamilies</a></li>
<li class="toctree-l1"><a class="reference internal" href="building.html">Building chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Structural analysis of chimeras</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#1.-Hydrophobic-cluster-analysis">1. Hydrophobic cluster analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#2.-Salt-bridges-analysis">2. Salt bridges analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#3.-Hydrogen-Network-analysis">3. Hydrogen Network analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#4.-Further-structural-analysis">4. Further structural analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#4.1-Calculate-solvent-accessible-surface-area">4.1 Calculate solvent accessible surface area</a></li>
<li class="toctree-l3"><a class="reference internal" href="#4.2-Secondary-structure-for-every-residue">4.2 Secondary structure for every residue</a></li>
<li class="toctree-l3"><a class="reference internal" href="#4.3-Calculating-distance-matrices">4.3 Calculating distance matrices</a></li>
<li class="toctree-l3"><a class="reference internal" href="#4.4-Calculating-contact-orders">4.4 Calculating contact orders</a></li>
</ul>
</li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ProtLego</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Structural analysis of chimeras</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/structural.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Structural-analysis-of-chimeras">
<h1>Structural analysis of chimeras<a class="headerlink" href="#Structural-analysis-of-chimeras" title="Permalink to this headline">¶</a></h1>
<p>We have seen how to find fragments from the Fuzzle database and build chimeras with them. We have also evaluated their potential energy with several energy scoring functions. Now we are going to have a look at the structures of these chimeras in depth.</p>
<p>First import the protlego module.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;/agh/projects/noelia/chimAuto/&#39;</span><span class="p">)</span> <span class="c1"># Path where ProtLego folder is located</span>
<span class="kn">from</span> <span class="nn">protlego</span> <span class="k">import</span> <span class="o">*</span>
</pre></div>
</div>
</div>
<p>Fetch for example by the HitID from the Fuzzle-database, as explained in detail in the “Fetching from the Fuzzle database” section. We can have a look at the chimeras that we built in tutorial 3.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">myhit</span><span class="o">=</span> <span class="n">fetch_id</span><span class="p">(</span><span class="s1">&#39;hh207clusters&#39;</span><span class="p">,</span><span class="s1">&#39;6242020&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Let’s build all possible chimera-combinations from query and subject, as explained in detail in the “Building chimeras” section. The chimeras will be stored as a dictionary in the variable chimeras:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">a</span><span class="o">=</span><span class="n">Builder</span><span class="p">(</span><span class="n">myhit</span><span class="p">)</span>
<span class="n">aln</span><span class="o">=</span><span class="n">a</span><span class="o">.</span><span class="n">get_alignment</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">query</span><span class="p">,</span><span class="n">myhit</span><span class="o">.</span><span class="n">no</span><span class="p">)</span>
<span class="n">qpairs</span><span class="p">,</span> <span class="n">spairs</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">get_pairs</span><span class="p">(</span><span class="n">aln</span><span class="p">)</span>
<span class="n">qPDB</span><span class="p">,</span> <span class="n">sPDB</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">superimpose_structures</span><span class="p">(</span><span class="n">partial_alignment</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">chimeras</span><span class="o">=</span><span class="n">a</span><span class="o">.</span><span class="n">build_chimeras</span><span class="p">(</span><span class="n">partial_alignment</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-01-21 12:26:22,079 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=&#34;all&#34; option when reading the file.
2020-01-21 12:26:23,329 - protlego - INFO - Found the following non-standar residue: MSE
2020-01-21 12:26:23,382 - protlego - INFO - Found the following non-standar residue: MSE
2020-01-21 12:26:23,471 - protlego - INFO - Performing alignment 1 with TMalign
2020-01-21 12:26:26,566 - protlego - INFO - The RMSD between the fragments is 2.92824649810791 over 9 alpha carbons
2020-01-21 12:26:26,583 - protlego - INFO - Performing alignment 2 with TMalign
2020-01-21 12:26:26,659 - protlego - INFO - The RMSD between the fragments is 1.9147093296051025 over 16 alpha carbons
2020-01-21 12:26:26,675 - protlego - INFO - Performing alignment 3 with TMalign
2020-01-21 12:26:26,752 - protlego - INFO - The RMSD between the fragments is 1.1340450048446655 over 11 alpha carbons
2020-01-21 12:26:26,772 - protlego - INFO - Performing alignment 4 with TMalign
2020-01-21 12:26:26,851 - protlego - INFO - The RMSD between the fragments is 1.791361927986145 over 19 alpha carbons
2020-01-21 12:26:26,871 - protlego - INFO - Performing alignment 5 with TMalign
2020-01-21 12:26:26,950 - protlego - INFO - The RMSD between the fragments is 3.4196701049804688 over 18 alpha carbons
2020-01-21 12:26:26,970 - protlego - INFO - Performing alignment 6 with TMalign
2020-01-21 12:26:27,045 - protlego - INFO - The RMSD between the fragments is 0.019341103732585907 over 3 alpha carbons
2020-01-21 12:26:27,064 - protlego - INFO - Performing alignment 7 with TMalign
2020-01-21 12:26:27,148 - protlego - INFO - The RMSD between the fragments is 1.3557014465332031 over 11 alpha carbons
2020-01-21 12:26:27,168 - protlego - INFO - Performing alignment 8 with TMalign
2020-01-21 12:26:27,267 - protlego - INFO - The RMSD between the fragments is 3.035205364227295 over 24 alpha carbons
</pre></div></div>
</div>
<div class="section" id="1.-Hydrophobic-cluster-analysis">
<h2>1. Hydrophobic cluster analysis<a class="headerlink" href="#1.-Hydrophobic-cluster-analysis" title="Permalink to this headline">¶</a></h2>
<p>The side chains of isoleucine, leucine, and valine (ILV) residues often form large hydrophobic clusters that very effectively impede the penetration of water to their underlying hydrogen bond networks and, thereby, enables these ILV clusters to serve as cores of stability in high‐energy partially folded states. For more information, see this publication: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/full/10.1002/pro.2860">https://onlinelibrary.wiley.com/doi/full/10.1002/pro.2860</a> To compute hydrophobic clusters for a chimera object, we use the
compute_hydrophobic_clusters() function. The ouput will be for each computed cluster:</p>
<p>The cluster index, the residues involved in the cluster formation, the total area, the number of contacts, the contacts per residues, and the area per residue.</p>
<p>The output can be stored in a variable to print it out for example write it into a file.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">clusters</span><span class="o">=</span> <span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">compute_hydrophobic_clusters</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">area</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">residues</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">contacts</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">ratio_contacts_residue</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">ratio_area_residue</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>With the view function one can visualize the clusters using vmd. They are displayed with VDW-spheres in different colors.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<p>There is also the possibility to change the selection for computing the clusters.The default selection are the hydrophobic residues Valine, Leucine and Isoleucine. For example, we can add PHE to the selection.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">compute_hydrophobic_clusters</span><span class="p">(</span><span class="n">sel</span><span class="o">=</span><span class="s1">&#39;protein and noh and not backbone and resname PHE LEU VAL ILE&#39;</span><span class="p">)</span>
<span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<p>Instead of a chimera object one can also work with a pdb initialized as a chimera object. Let’s use the pdb from HisF, the pdb we used for the first VMD practical</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">protein</span><span class="o">=</span><span class="n">Chimera</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s1">&#39;/agh/projects/noelia/abhishek/chimera/2lle.pdb&#39;</span><span class="p">)</span>
<span class="n">protein</span><span class="o">.</span><span class="n">compute_hydrophobic_clusters</span><span class="p">()</span>
<span class="n">protein</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="2.-Salt-bridges-analysis">
<h2>2. Salt bridges analysis<a class="headerlink" href="#2.-Salt-bridges-analysis" title="Permalink to this headline">¶</a></h2>
<p>A salt bridge is a combination of two non-covalent interactions: hydrogen bonding and ionic bonding. Ion pairing is one of the most important noncovalent forces in biological systems and it contributes to the stability to the entropically unfavorable folded conformation of proteins.</p>
<p>To compute the salt bridges for a chimera object use the compute_salt_bridges() function.</p>
<p>Use the view() function to visualize the salt bridges they will be displayed in Licorice. The positively charged residues are displayed in blue and the negatively charged residue are displayed in red. As before, one can also work with a pdb initialized as a chimera object.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">protein</span><span class="o">=</span><span class="n">Chimera</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s1">&#39;/agh/projects/noelia/abhishek/chimera/2lle.pdb&#39;</span><span class="p">)</span>
<span class="n">protein</span><span class="o">.</span><span class="n">compute_salt_bridges</span><span class="p">()</span>
<span class="n">protein</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="3.-Hydrogen-Network-analysis">
<h2>3. Hydrogen Network analysis<a class="headerlink" href="#3.-Hydrogen-Network-analysis" title="Permalink to this headline">¶</a></h2>
<p>A hydrogen bond (H-bond) is a primarily electrostatic force of attraction between a hydrogen (H) atom which is covalently bound to a more electronegative atom (called the donor) and another electronegative atom bearing a lone pair of electrons—the hydrogen bond acceptor (called the acceptor)</p>
<p>To compute the hydrogen networks for a chimera object use the compute_hydrogen_networks() function. With the sidechain_only argument it is possible to compute hydrogen networks not only between the sidechains, which is the default.</p>
<p>Use the view() function to visualize the hydrogen bonds they are displayed in licorice. They are represented in different colors when they belong to the same hydrogen bond network. Again one can also work with a pdb initialized as a chimera object.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">protein</span><span class="o">=</span><span class="n">Chimera</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s1">&#39;/agh/projects/noelia/abhishek/chimera/2lle.pdb&#39;</span><span class="p">)</span>
<span class="n">protein</span><span class="o">.</span><span class="n">compute_hydrogen_networks</span><span class="p">()</span>
<span class="n">protein</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-01-21 12:25:16,046 - propka - INFO - No pdbfile provided
/agh/projects/noelia/miniconda3/envs/my-rdkit-env/lib/python3.6/site-packages/numpy/core/fromnumeric.py:57: FutureWarning: Series.nonzero() is deprecated and will be removed in a future version.Use Series.to_numpy().nonzero() instead
  return getattr(obj, method)(*args, **kwds)
</pre></div></div>
</div>
</div>
<div class="section" id="4.-Further-structural-analysis">
<h2>4. Further structural analysis<a class="headerlink" href="#4.-Further-structural-analysis" title="Permalink to this headline">¶</a></h2>
<div class="section" id="4.1-Calculate-solvent-accessible-surface-area">
<h3>4.1 Calculate solvent accessible surface area<a class="headerlink" href="#4.1-Calculate-solvent-accessible-surface-area" title="Permalink to this headline">¶</a></h3>
<p>The solvent-accessible surface area (SASA) is the surface area of a biomolecule that is accessible to a solvent. The function calc_sasa() calculates for a chimera object the solvent accessible surface area. By default, the function computes the total SASA, but it is also possible to calculate the polar or apolar surface area with the parameter sasa_type=“polar”, sasa_type=“apolar”.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">total</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">])</span>
<span class="n">polar</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">]</span> <span class="p">,</span><span class="n">sasa_type</span><span class="o">=</span><span class="s2">&quot;polar&quot;</span><span class="p">)</span>
<span class="n">apolar</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">]</span> <span class="p">,</span><span class="n">sasa_type</span><span class="o">=</span><span class="s2">&quot;apolar&quot;</span><span class="p">)</span>
<span class="n">totalpdbfile</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s1">&#39;/agh/projects/noelia/abhishek/chimera/2lle.pdb&#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-01-21 12:28:29,533 - protlego - INFO - Area is 120.56597900390625 (nm)^2
2020-01-21 12:28:29,788 - protlego - INFO - Area is 94.2850341796875 (nm)^2
2020-01-21 12:28:30,033 - protlego - INFO - Area is 24.696435928344727 (nm)^2
/agh/projects/noelia/miniconda3/envs/my-rdkit-env/lib/python3.6/site-packages/mdtraj/formats/pdb/pdbfile.py:196: UserWarning: Unlikely unit cell vectors detected in PDB file likely resulting from a dummy CRYST1 record. Discarding unit cell vectors.
  warnings.warn(&#39;Unlikely unit cell vectors detected in PDB file likely &#39;
2020-01-21 12:28:32,717 - protlego - INFO - Area is 126.6241455078125 (nm)^2
</pre></div></div>
</div>
</div>
<div class="section" id="4.2-Secondary-structure-for-every-residue">
<h3>4.2 Secondary structure for every residue<a class="headerlink" href="#4.2-Secondary-structure-for-every-residue" title="Permalink to this headline">¶</a></h3>
<p>To calculate the secondary structure for every residue in a pdb or a chimera object we can use the function calc_dssp(). The output will be an array containg a one letter code for the secondary structure for each residue.</p>
<ul class="simple">
<li><p>C: Coil</p></li>
<li><p>H: Helix</p></li>
<li><p>E: Sheet</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">calc_dssp</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>array([[&#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;]],
      dtype=&#39;&lt;U2&#39;)
</pre></div>
</div>
</div>
<p>If the parameter simplified is set to false. The code expandes:</p>
<ul class="simple">
<li><p>H: Alpha helix</p></li>
<li><p>B: Isolated beta-bridge residue</p></li>
<li><p>E: Strand</p></li>
<li><p>G: 3-10 helix</p></li>
<li><p>I: Pi helix</p></li>
<li><p>T: Turn</p></li>
<li><p>S: Bend</p></li>
<li><p>’ ’: None</p></li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">calc_dssp</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">],</span> <span class="n">simplified</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>array([[&#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;S&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;S&#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;,
        &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;H&#39;, &#39;I&#39;, &#39;I&#39;, &#39;I&#39;, &#39;I&#39;,
        &#39;I&#39;, &#39;I&#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;S&#39;, &#39;S&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;S&#39;, &#39;T&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39; &#39;, &#39;S&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39; &#39;, &#39;S&#39;, &#39;S&#39;, &#39;S&#39;, &#39; &#39;, &#39; &#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;G&#39;, &#39;G&#39;, &#39;G&#39;, &#39;T&#39;,
        &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;S&#39;, &#39;S&#39;, &#39; &#39;, &#39; &#39;, &#39;S&#39;, &#39; &#39;, &#39; &#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;G&#39;, &#39;G&#39;, &#39;G&#39;, &#39;G&#39;, &#39;G&#39;, &#39;G&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;T&#39;, &#39;S&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;S&#39;, &#39;S&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39; &#39;, &#39; &#39;]],
      dtype=&#39;&lt;U2&#39;)
</pre></div>
</div>
</div>
</div>
<div class="section" id="4.3-Calculating-distance-matrices">
<h3>4.3 Calculating distance matrices<a class="headerlink" href="#4.3-Calculating-distance-matrices" title="Permalink to this headline">¶</a></h3>
<p>The function calc_dist_matrix calculates a distance matrix for a chimera object or a pdb file. The selection can be defined, default selection is by residue but selection can be also alpha for taking only the C alpha atoms into account. Also the type can be defined default is contacts for calculating only a contact map, but the parameter type can be also set to distances to draw a distance matrix. The parameter plot can be set to true to draw a plot.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">contact_map</span><span class="o">=</span><span class="n">calc_dist_matrix</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">],</span> <span class="n">plot</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">distance_matrix</span><span class="o">=</span><span class="n">calc_dist_matrix</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">],</span><span class="nb">type</span><span class="o">=</span><span class="s1">&#39;distances&#39;</span><span class="p">,</span> <span class="n">plot</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/structural_36_0.png" src="_images/structural_36_0.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/structural_36_1.png" src="_images/structural_36_1.png" />
</div>
</div>
</div>
<div class="section" id="4.4-Calculating-contact-orders">
<h3>4.4 Calculating contact orders<a class="headerlink" href="#4.4-Calculating-contact-orders" title="Permalink to this headline">¶</a></h3>
<p>The function calc_contact_order can be used to calculate the contact order of a protein. It is a measure of the locality of the inter-amino acid contacts in the native folded state. It is computed as the average sequence distance between residues that form contacts below a threshold in the folded protein divided by the total length of the protein. The function can calculate the contact order of a chimera object or a pdb file and the distance cutoff can be set, the default distance cutoff is 8
Ångström. The output is the contact order.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">calc_contact_order</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_103&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Contact order is 14.391958203440714 %
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="output_area highlight-none notranslate"><div class="highlight"><pre>
<span></span>14.391958203440714
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="building.html" class="btn btn-neutral float-left" title="Building chimeras" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Noelia Ferruz

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>