<!DOCTYPE html>
<html class="writer-html4" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Structural analysis of chimeras &mdash; ProtLego 2.0.0 documentation</title><link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/nbsphinx-code-cells.css" type="text/css" />
      <link rel="stylesheet" href="_static/nbsphinx-code-cells.css" type="text/css" />
      <link rel="stylesheet" href="_static/nbsphinx-code-cells.css" type="text/css" />
      <link rel="stylesheet" href="_static/nbsphinx-code-cells.css" type="text/css" />
      <link rel="stylesheet" href="_static/nbsphinx-code-cells.css" type="text/css" />
      <link rel="stylesheet" href="_static/nbsphinx-code-cells.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "tex2jax_ignore|mathjax_ignore|document", "processClass": "tex2jax_process|mathjax_process|math|output_area"}})</script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Energy minimization" href="energy_min.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            ProtLego
              <img src="_static/logo.png" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                2.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Code</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="builder.html">Building Chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="builder.html#chimera-class">Chimera Class</a></li>
<li class="toctree-l1"><a class="reference internal" href="database.html">Fetching hits from Fuzzle</a></li>
<li class="toctree-l1"><a class="reference internal" href="networks.html">Networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy.html">Energy scoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html">Clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#hh-networks">HH-networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#module-structural.salt_bridges">Salt-bridges</a></li>
<li class="toctree-l1"><a class="reference internal" href="analyse.html#further-structural-analysis">Further structural analysis</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction to Jupyter-notebooks and Python primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="fetching.html">Fetching from the Fuzzle database</a></li>
<li class="toctree-l1"><a class="reference internal" href="network.html">Drawing networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="building.html">Building chimeras</a></li>
<li class="toctree-l1"><a class="reference internal" href="energy_min.html">Energy minimization</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Structural analysis of chimeras</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#1.-Hydrophobic-cluster-analysis">1. Hydrophobic cluster analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#2.-Salt-bridges-analysis">2. Salt bridges analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#3.-Hydrogen-Network-analysis">3. Hydrogen Network analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#4.-Further-structural-analysis">4. Further structural analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#4.1-Calculate-solvent-accessible-surface-area">4.1 Calculate solvent accessible surface area</a></li>
<li class="toctree-l3"><a class="reference internal" href="#4.2-Secondary-structure-for-every-residue">4.2 Secondary structure for every residue</a></li>
<li class="toctree-l3"><a class="reference internal" href="#4.3-Calculating-distance-matrices">4.3 Calculating distance matrices</a></li>
<li class="toctree-l3"><a class="reference internal" href="#4.4-Calculating-contact-orders">4.4 Calculating contact orders</a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ProtLego</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Structural analysis of chimeras</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/structural.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="Structural-analysis-of-chimeras">
<h1>Structural analysis of chimeras<a class="headerlink" href="#Structural-analysis-of-chimeras" title="Permalink to this headline">¶</a></h1>
<p>We have seen how to find fragments from the Fuzzle database and build chimeras with them. We have also evaluated their potential energy with several energy scoring functions. Now we are going to have a look at the structures of these chimeras in depth.</p>
<p>First import the protlego module.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">sys</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;YOUR_PATH_TO_PROTLEGO_FOLDER&quot;</span><span class="p">)</span>

<span class="kn">from</span> <span class="nn">protlego.all</span> <span class="k">import</span> <span class="o">*</span>
</pre></div>
</div>
</div>
<p>Fetch for example by the HitID from the Fuzzle-database, as explained in detail in the “Fetching from the Fuzzle database” section. We can have a look at the chimeras that we built in tutorial 3.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">myhit</span><span class="o">=</span> <span class="n">fetch_id</span><span class="p">(</span><span class="s1">&#39;4413706&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Let’s build all possible chimera-combinations from query and subject, as explained in detail in the “Building chimeras” section. The chimeras will be stored as a dictionary in the variable chimeras:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">a</span><span class="o">=</span><span class="n">Builder</span><span class="p">(</span><span class="n">myhit</span><span class="p">)</span>
<span class="n">aln</span><span class="o">=</span><span class="n">a</span><span class="o">.</span><span class="n">get_alignment</span><span class="p">(</span><span class="n">myhit</span><span class="o">.</span><span class="n">query</span><span class="p">,</span><span class="n">myhit</span><span class="o">.</span><span class="n">no</span><span class="p">)</span>
<span class="n">qPDB</span><span class="p">,</span> <span class="n">sPDB</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">superimpose_structures</span><span class="p">(</span><span class="n">aln</span><span class="p">,</span><span class="n">partial_alignment</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">chimeras</span><span class="o">=</span><span class="n">a</span><span class="o">.</span><span class="n">build_chimeras</span><span class="p">(</span><span class="n">partial_alignment</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-12 15:49:46,155 - protlego - INFO - Attempting to download domain d2dfda1 from the SCOP server
2020-10-12 15:49:47,510 - protlego - INFO - File downloaded as /tmp/tmpc55d_xpa.pdb
2020-10-12 15:49:47,511 - protlego - INFO - Attempting to download domain d1wa5a_ from the SCOP server
2020-10-12 15:49:49,156 - protlego - INFO - File downloaded as /tmp/tmpbnqv9qmo.pdb
2020-10-12 15:49:49,157 - protlego - INFO - Loading /tmp/tmpc55d_xpa.pdb as a chimera object
2020-10-12 15:49:49,239 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=&#34;all&#34; option when reading the file.
2020-10-12 15:49:49,241 - protlego - INFO - Loading /tmp/tmpbnqv9qmo.pdb as a chimera object
2020-10-12 15:49:49,302 - protlego - INFO - Attempting to download hhs file for d2dfda1 from the FUZZLE server
2020-10-12 15:49:49,435 - protlego - INFO - File downloaded as /tmp/tmpnxx_3e1q.hhs
2020-10-12 15:49:50,463 - protlego - INFO - Performing alignment 1 with TMalign
2020-10-12 15:49:53,556 - protlego - INFO - The RMSD between the fragments is 0.6881080269813538 over 6 alpha carbons
2020-10-12 15:49:53,564 - protlego - INFO - Performing alignment 2 with TMalign
2020-10-12 15:49:53,629 - protlego - INFO - The RMSD between the fragments is 2.3234670162200928 over 6 alpha carbons
2020-10-12 15:49:53,637 - protlego - INFO - Performing alignment 3 with TMalign
2020-10-12 15:49:53,705 - protlego - INFO - The RMSD between the fragments is 1.1804629564285278 over 9 alpha carbons
2020-10-12 15:49:53,712 - protlego - INFO - Performing alignment 4 with TMalign
2020-10-12 15:49:53,792 - protlego - INFO - The RMSD between the fragments is 10.348335266113281 over 34 alpha carbons
2020-10-12 15:49:53,802 - protlego - INFO - Performing alignment 5 with TMalign
2020-10-12 15:49:53,874 - protlego - INFO - The RMSD between the fragments is 4.646137714385986 over 17 alpha carbons
2020-10-12 15:49:53,883 - protlego - INFO - Performing alignment 6 with TMalign
2020-10-12 15:49:53,962 - protlego - INFO - The RMSD between the fragments is 3.52919340133667 over 29 alpha carbons
</pre></div></div>
</div>
<div class="section" id="1.-Hydrophobic-cluster-analysis">
<h2>1. Hydrophobic cluster analysis<a class="headerlink" href="#1.-Hydrophobic-cluster-analysis" title="Permalink to this headline">¶</a></h2>
<p>The side chains of isoleucine, leucine, and valine (ILV) residues often form large hydrophobic clusters that very effectively impede the penetration of water to their underlying hydrogen bond networks. These ILV clusters serve as cores of stability in high‐energy partially folded states. For more information, see this publication: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/full/10.1002/pro.2860">https://onlinelibrary.wiley.com/doi/full/10.1002/pro.2860</a> To compute hydrophobic clusters for a chimera object, we use the compute_hydrophobic_clusters() function. The
ouput will be for each computed cluster:</p>
<ul class="simple">
<li>the cluster index,</li>
<li>the residues involved in the cluster formation,</li>
<li>the total area,</li>
<li>the number of contacts,</li>
<li>the contacts per residues, and</li>
<li>the area per residue.</li>
</ul>
<p>The output can be stored in a variable to print it out for example write it into a file.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">clusters</span><span class="o">=</span> <span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">compute_hydrophobic_clusters</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-12 15:53:00,434 - protlego - INFO - Initializing final output
2020-10-12 15:53:00,438 - protlego - INFO - Computing clusters
2020-10-12 15:53:03,739 - hydrophobic_clusters - INFO - Cluster index 1:Residues 17. Total area 2120.7734167837134 A^2. Number of contacts: 52. Contacts / Residue: 3.0588235294117645. Area / Residue 40.784104168917565
2020-10-12 15:53:03,742 - hydrophobic_clusters - INFO - Cluster index 2:Residues 12. Total area 1748.6573579708954 A^2. Number of contacts: 35. Contacts / Residue: 2.9166666666666665. Area / Residue 49.96163879916844
2020-10-12 15:53:03,746 - hydrophobic_clusters - INFO - Cluster index 4:Residues 3. Total area 104.60916007126269 A^2. Number of contacts: 4. Contacts / Residue: 1.3333333333333333. Area / Residue 26.152290017815673
2020-10-12 15:53:03,749 - hydrophobic_clusters - INFO - Cluster index 8:Residues 2. Total area 27.260437899926508 A^2. Number of contacts: 2. Contacts / Residue: 1.0. Area / Residue 13.630218949963254
2020-10-12 15:53:03,752 - hydrophobic_clusters - INFO - Cluster index 11:Residues 2. Total area 100.6198276956637 A^2. Number of contacts: 2. Contacts / Residue: 1.0. Area / Residue 50.30991384783185
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Cluster(area=2120.7734167837134, residues=[3, 5, 12, 16, 18, 20, 26, 29, 31, 40, 44, 47, 53, 79, 110, 141, 156], contacts=52, ratio_contacts_residue=3.0588235294117645, ratio_area_residue=40.784104168917565)
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">area</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">residues</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">contacts</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">ratio_contacts_residue</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">clusters</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">ratio_area_residue</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
2120.77341678
[3, 5, 12, 16, 18, 20, 26, 29, 31, 40, 44, 47, 53, 79, 110, 141, 156]
52
3.0588235294117645
40.7841041689
</pre></div></div>
</div>
<p>With the view function one can visualize the clusters using vmd. They are displayed with VDW-spheres in different colors.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<p>There is also the possibility to change the selection for computing the clusters.The default selection are the hydrophobic residues Valine, Leucine and Isoleucine. For example, we can add PHE to the selection.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">compute_hydrophobic_clusters</span><span class="p">(</span><span class="n">sel</span><span class="o">=</span><span class="s1">&#39;protein and noh and not backbone and resname PHE LEU VAL ILE&#39;</span><span class="p">)</span>
<span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-12 15:53:29,968 - protlego - INFO - Initializing final output
2020-10-12 15:53:29,972 - protlego - INFO - Computing clusters
2020-10-12 15:53:34,019 - hydrophobic_clusters - INFO - Cluster index 1:Residues 25. Total area 4122.088491875879 A^2. Number of contacts: 74. Contacts / Residue: 2.96. Area / Residue 55.703898538863236
2020-10-12 15:53:34,024 - hydrophobic_clusters - INFO - Cluster index 2:Residues 18. Total area 2627.861887637631 A^2. Number of contacts: 54. Contacts / Residue: 3.0. Area / Residue 48.6641090303265
</pre></div></div>
</div>
<p>Instead of chimeras, one could also work with a standard pdb, by initializing it as a chimera object.</p>
</div>
<div class="section" id="2.-Salt-bridges-analysis">
<h2>2. Salt bridges analysis<a class="headerlink" href="#2.-Salt-bridges-analysis" title="Permalink to this headline">¶</a></h2>
<p>A salt bridge is a combination of two non-covalent interactions: hydrogen bonding and ionic bonding. Ion pairing is one of the most important noncovalent forces in biological systems and it contributes to the stability to the entropically unfavorable folded conformation of proteins.</p>
<p>To compute the salt bridges for a chimera object use the compute_salt_bridges() function.</p>
<p>Use the view() function to visualize the salt bridges they will be displayed in Licorice. The positively charged residues are displayed in blue and the negatively charged residue are displayed in red. As before, one can also work with a pdb initialized as a chimera object.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span>
<span class="n">chimera</span><span class="o">.</span><span class="n">compute_salt_bridges</span><span class="p">()</span>
<span class="n">chimera</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
563 823
1041 873
1265 1196
</pre></div></div>
</div>
</div>
<div class="section" id="3.-Hydrogen-Network-analysis">
<h2>3. Hydrogen Network analysis<a class="headerlink" href="#3.-Hydrogen-Network-analysis" title="Permalink to this headline">¶</a></h2>
<p>A hydrogen bond (H-bond) is a primarily electrostatic force of attraction between a hydrogen (H) atom which is covalently bound to a more electronegative atom (called the donor) and another electronegative atom bearing a lone pair of electrons—the hydrogen bond acceptor (called the acceptor)</p>
<p>To compute the hydrogen networks for a chimera object use the compute_hydrogen_networks() function. With the sidechain_only argument it is possible to compute hydrogen networks not only between the sidechains, which is the default.</p>
<p>Use the view() function to visualize the hydrogen bonds they are displayed in licorice. They are represented in different colors when they belong to the same hydrogen bond network. Again one can also work with a pdb initialized as a chimera object.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">chimera</span><span class="o">.</span><span class="n">compute_hydrogen_networks</span><span class="p">()</span>
<span class="n">chimera</span><span class="o">.</span><span class="n">view</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-01-21 12:25:16,046 - propka - INFO - No pdbfile provided
/agh/projects/noelia/miniconda3/envs/my-rdkit-env/lib/python3.6/site-packages/numpy/core/fromnumeric.py:57: FutureWarning: Series.nonzero() is deprecated and will be removed in a future version.Use Series.to_numpy().nonzero() instead
  return getattr(obj, method)(*args, **kwds)
</pre></div></div>
</div>
</div>
<div class="section" id="4.-Further-structural-analysis">
<h2>4. Further structural analysis<a class="headerlink" href="#4.-Further-structural-analysis" title="Permalink to this headline">¶</a></h2>
<div class="section" id="4.1-Calculate-solvent-accessible-surface-area">
<h3>4.1 Calculate solvent accessible surface area<a class="headerlink" href="#4.1-Calculate-solvent-accessible-surface-area" title="Permalink to this headline">¶</a></h3>
<p>The solvent-accessible surface area (SASA) is the surface area of a biomolecule that is accessible to a solvent. The function calc_sasa() calculates for a chimera object the solvent accessible surface area. By default, the function computes the total SASA, but it is also possible to calculate the polar or apolar surface area with the parameter sasa_type=”polar”, sasa_type=”apolar”.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">total</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">])</span>
<span class="n">polar</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span> <span class="p">,</span><span class="n">sasa_type</span><span class="o">=</span><span class="s2">&quot;polar&quot;</span><span class="p">)</span>
<span class="n">apolar</span><span class="o">=</span><span class="n">calc_sasa</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">]</span> <span class="p">,</span><span class="n">sasa_type</span><span class="o">=</span><span class="s2">&quot;apolar&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
2020-10-12 16:04:08,365 - protlego - INFO - Area is 87.79385375976562 (nm)^2
2020-10-12 16:04:08,526 - protlego - INFO - Area is 60.05790710449219 (nm)^2
2020-10-12 16:04:08,699 - protlego - INFO - Area is 27.735942840576172 (nm)^2
</pre></div></div>
</div>
</div>
<div class="section" id="4.2-Secondary-structure-for-every-residue">
<h3>4.2 Secondary structure for every residue<a class="headerlink" href="#4.2-Secondary-structure-for-every-residue" title="Permalink to this headline">¶</a></h3>
<p>To calculate the secondary structure for every residue in a pdb or a chimera object we can use the function calc_dssp(). The output will be an array containg a one letter code for the secondary structure for each residue.</p>
<ul class="simple">
<li>C: Coil</li>
<li>H: Helix</li>
<li>E: Sheet</li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">calc_dssp</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
array([[&#39;C&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;C&#39;,
        &#39;C&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;, &#39;C&#39;]],
      dtype=&#39;&lt;U2&#39;)
</pre></div></div>
</div>
<p>If the parameter simplified is set to false. The code expandes:</p>
<ul class="simple">
<li>H: Alpha helix</li>
<li>B: Isolated beta-bridge residue</li>
<li>E: Strand</li>
<li>G: 3-10 helix</li>
<li>I: Pi helix</li>
<li>T: Turn</li>
<li>S: Bend</li>
<li>‘ ‘: None</li>
</ul>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">calc_dssp</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">],</span> <span class="n">simplified</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
array([[&#39; &#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;S&#39;, &#39;T&#39;, &#39;T&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;S&#39;, &#39;S&#39;, &#39;S&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;S&#39;, &#39;S&#39;, &#39;S&#39;, &#39;S&#39;, &#39; &#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;S&#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39; &#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;,
        &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;S&#39;, &#39;S&#39;, &#39;S&#39;, &#39; &#39;, &#39; &#39;, &#39;E&#39;,
        &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39;S&#39;, &#39;S&#39;, &#39; &#39;, &#39; &#39;, &#39;S&#39;,
        &#39;S&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39; &#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;T&#39;,
        &#39; &#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39;E&#39;, &#39; &#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;,
        &#39;T&#39;, &#39;T&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;, &#39;H&#39;,
        &#39;T&#39;, &#39; &#39;, &#39;T&#39;, &#39;T&#39;, &#39; &#39;, &#39; &#39;, &#39; &#39;]],
      dtype=&#39;&lt;U2&#39;)
</pre></div></div>
</div>
</div>
<div class="section" id="4.3-Calculating-distance-matrices">
<h3>4.3 Calculating distance matrices<a class="headerlink" href="#4.3-Calculating-distance-matrices" title="Permalink to this headline">¶</a></h3>
<p>The function calc_dist_matrix calculates a distance matrix for a chimera object or a pdb file. The selection can be defined, default selection is by residue but selection can be also alpha for taking only the C alpha atoms into account. Also the type can be defined default is contacts for calculating only a contact map, but the parameter type can be also set to distances to draw a distance matrix. The parameter plot can be set to true to draw a plot.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">contact_map</span><span class="o">=</span><span class="n">calc_dist_matrix</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">],</span> <span class="n">plot</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">distance_matrix</span><span class="o">=</span><span class="n">calc_dist_matrix</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">],</span><span class="nb">type</span><span class="o">=</span><span class="s1">&#39;distances&#39;</span><span class="p">,</span> <span class="n">plot</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/structural_35_0.png" src="_images/structural_35_0.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/structural_35_1.png" src="_images/structural_35_1.png" />
</div>
</div>
</div>
<div class="section" id="4.4-Calculating-contact-orders">
<h3>4.4 Calculating contact orders<a class="headerlink" href="#4.4-Calculating-contact-orders" title="Permalink to this headline">¶</a></h3>
<p>The function calc_contact_order can be used to calculate the contact order of a protein. It is a measure of the locality of the inter-amino acid contacts in the native folded state. It is computed as the average sequence distance between residues that form contacts below a threshold in the folded protein divided by the total length of the protein. The function can calculate the contact order of a chimera object or a pdb file and the distance cutoff can be set, the default distance cutoff is 8
Ångström. The output is the contact order.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">calc_contact_order</span><span class="p">(</span><span class="n">chimera</span><span class="o">=</span><span class="n">chimeras</span><span class="p">[</span><span class="s1">&#39;comb1_72&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Contact order is 16.048235723913784 %
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
16.048235723913784
</pre></div></div>
</div>
</div>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="energy_min.html" class="btn btn-neutral float-left" title="Energy minimization" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2020, Noelia Ferruz.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>